tests/pytests/l3_dump_test.py

```diff
@@ -0,0 +1,545 @@
+# #############################################################################
+# l3_dump_test.py
+#
+"""
+Collection of basic test cases to exercise different combinations of `make`
+commands. Cross-check that the output from the L3 Python dump script
+comes out as expected. Verify, either using hard-coded expected outputs,
+or outputs generated by using `grep` on source-files, that the outputs
+returned by l3_dump.py script are as per the expected outputs.
+"""
+import os
+import sys
+import subprocess as sp
+import shlex
+from fnmatch import fnmatchcase
+
+# #############################################################################
+# Setup some variables pointing to diff dir/sub-dir full-paths.
+# Dir-tree:
+#  /tests/pytests/
+#   - <this-file>
+# Full dir-path where this tests/  dir lives
+L3PytestsDir    = os.path.realpath(os.path.dirname(__file__))
+L3RootDir       = os.path.realpath(L3PytestsDir + '/../..')
+L3USE_CASES_DIR = L3RootDir + '/use-cases/'
+
+sys.path.append(L3RootDir)
+
+# pylint: disable-msg=import-error,wrong-import-position
+import l3_dump
+
+L3DUMPSCRIPT    = 'l3_dump.py'
+L3Dump          = L3RootDir + '/' + L3DUMPSCRIPT
+
+L3_LOC_EXPLICITLY_UNSET = "0"
+L3_LOC_DEFAULT          = "1"
+L3_LOC_ELF_ENCODING     = "2"
+
+# Ternary: variable = something if condition else something_else
```

Comment:
is this comment supposed to be here?

Reply:
Yes; I left behind as I continually find myself searching for this syntax, so left a small comment in-there.